mbt_gym/agents/BaselineAgents.py

```diff
@@ -146,3 +147,37 @@ def _calculate_a_and_z(self):
             if i > 0:
                 Amatrix[i, i - 1] = self.lambdas[1] * np.exp(-1)
         return Amatrix, z_vector
+
+
+class CarteaJaimungalOeAgent(Agent):
+    def __init__(
+        self,
+        phi: float = 2 * 10 ** (-4),
+        alpha: float = 0.0001,
+        env: TradingEnvironment = None,
+    ):
+        self.phi = phi
+        self.alpha = alpha
+        self.env = env or TradingEnvironment()
+        self.price_impact_model = env.price_impact_model
+        assert self.env.action_type == "speed"
+        self.terminal_time = self.env.terminal_time
+        self.temporary_price_impact = self.price_impact_model.temporary_impact_coefficient
+        self.permanent_price_impact = self.price_impact_model.permanent_impact_coefficient
+        self.num_trajectories = self.env.num_trajectories
+
+    def get_action(self, state: np.ndarray):
+        action = np.zeros(shape=(self.num_trajectories, 1))
+        # The formulae below is in page 146 of Cartea, Jaimungal, Penalva (2015) 
+        # Algorithmic and High-Frequency Trading
+        # Cambridge University Press
+        gamma = np.sqrt(self.phi/self.temporary_price_impact)
+        zeta = (self.alpha - 0.5 * self.permanent_price_impact + np.sqrt(self.temporary_price_impact * self.phi))
+        zeta /= (self.alpha - 0.5 * self.permanent_price_impact - np.sqrt(self.temporary_price_impact * self.phi))  
+        initial_inventory = self.env.initial_inventory
+        for iq, q in enumerate(state[:, INVENTORY_INDEX]):
```

Comment:
I think this should be accelerated to assume that time is **always** the same for a given state. This will make the CJ Agent muuuuuch faster when considering a vectorised env.

Reply:
(using numpy directly)